### PR TITLE
Added ability to retrace Singularity hub urls

### PR DESCRIPTION
**Pull request by mjtravers**
Commit addresses issue #263 

TODOs
- [x] add `path` field to the SingularityImage class
- [x] install image if `url` is known, if also `path` is known -- pull into that location

### CODE REVIEW

File: niceman/distributions/tests/test_singularity.py [hunk 1]
```diff
@@ -46,3 +46,16 @@ def test_singularity_trace():
 
     # Clean up test files
     os.remove(image_file)
+
+    # Run tracer against a Singularity Hub url
+    files = ['shub:/GodloveD/lolcow']
```
Comment (kyleam) [inline niceman/distributions/tests/test_singularity.py]: If this image is updated, this test will start to fail.  Pulling with the GitHub commit (appending "@a59d8de3121579fe9c95ab8af0297c2e3aefd827") would avoid that.

File: niceman/distributions/singularity.py [hunk 2]
```diff
@@ -95,14 +100,30 @@ def identify_distributions(self, files):
...
         for file_path in files:
             try:
-                image = json.loads(self._session.execute_command(['singularity',
-                    'inspect', file_path])[0])
+                if file_path.startswith('shub:/'):
+                    # Correct file path for path normalization in retrace.py
+                    if not file_path.startswith('shub://'):
+                        file_path = file_path.replace('shub:/', 'shub://')
+                    temp_path = "{}.simg".format(uuid.uuid4())
+                    with chpwd(tempfile.gettempdir()):
```
Comment (yarikoptic) [inline niceman/distributions/singularity.py]: I think there should be some log message (lgr.info) to announce that we will download the thing to be "traced"

**Comment by yarikoptic** [comment]
please
- [ ] add `path` field to the SingularityImage class
- [ ] install image if `url` is known, if also `path` is known -- pull into that location

**Comment by yarikoptic** [comment]
see https://github.com/singularityware/singularity/issues/1775#issuecomment-406655734 about url to specify exact version

**Comment by mjtravers** [comment]
All comments have been resolved.

**Pull request merged**